agent/lib/events.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var util = require('util');
+
+
+function hash(s) {
+    var h = 0;
+    for (var i = 0; i < s.length; i++) {
+        h = (h << 5) - h + s.charCodeAt(i) | 0;
+    }
+    return h;
+};
+
+
+function event (component, reason, message, type, kind, name) {
+    var event_id = util.format('%s.%s', component, hash([kind, reason, name, message].join()));
+    var now = new Date().toISOString();
+    return {
+        kind: 'Event',
+        metadata: {
+            name: event_id
+        },
+        count: 1,
+        reason: reason,
+        message: message,
+        type: type,
+        firstTimestamp: now,
+        lastTimestamp: now,
+        source: {
+            component: component
+        },
+        involvedObject: {
+            kind: kind,
+            name: name
+        }
+    };
+}
+
+module.exports.address_create = function (address) {
+    return event('agent', 'Create', util.format('%s %s created', address.type, address.address), 'Normal', 'Address', address.address);
```

Comment:
I don't see any other events that repeat the kind in the event. Seems a little redundant to me. Obviously where there are different things created/deleted with the same involved object, that does require some qualification. Not a huge issue though, so I could live with AddressCreated/AddressDeleted.

It seems odd to me to have FailedAddressCreated though, FailedAddressCreate sounds clearer. ReplicaSet seems to use SuccessfulCreate (which would also match nicely with FailedCreate). I see now you are using AddressSpaceCreateFailed. The doc you link to above is itself inconsistent on this with a mix of pre- and post- fixing with failed (even in the same table!).

Bottom line though, I don't care too much. I agree we should try to be consistent (though in general there is no real consistency with existing events). Its pretty arbitrary I think. Since you already committed, I guess it makes sense to just go with that.